### ISSUE TITLE
Only one session in grouped session bells

### ISSUE THREAD

**Issue by MacGyverNL** [tmux/tmux#427]
Since commit b9563340b7e9f516f77e674791e7635e948ca7fa grouped sessions no longer have desired bell behaviour. When set to "bell-action current", and each session has a different window (or pane, but I have a 1-pane-per-window setup so haven't checked that) in the foreground, the desired behaviour is that if window 0 bells, then the session that has window 0 in the foreground will bell, and the session with window 1 will not; and vice versa. However, observed behaviour since this commit is that one session will bell on its foreground window, but the other will never bell.

Steps to reproduce:
- ```tmux new-session -s s1```
- ```tmux new-session -s s2 -t s1```
- In both sessions: ```:set bell-action current```
- In session s2: ```^b-c``` (new window)
- Ensure that s1 has window 1 on the foreground, s2 has window 2 on the foreground
- ```sleep 5 && printf "\007"``` in s1
- Ensure focus is on s2, see whether s1 bells or not.
- ```sleep 5 && printf "\007"``` in s2
- Ensure focus is on s1, see whether s2 bells or not.
- If s1 bells, s2 will not, and vice versa.

This holds regardless of whether the grouped session that doesn't bell is in focus, on a different workspace, on the same workspace but out of focus, etc.

**Comment by nicm** [comment]
Yes this is a regression I think.

**Comment by ThomasAdam** [comment]
I haven't tested this much, but does the following even go so far as to help with this problem?

```
diff --git a/alerts.c b/alerts.c
index 3667835..8231206 100644
--- a/alerts.c
+++ b/alerts.c
@@ -49,13 +49,22 @@ alerts_timer(__unused int fd, __unused short events, void *arg)
 void
 alerts_callback(__unused int fd, __unused short events, __unused void *arg)
 {
-       struct window   *w;
-       struct session  *s;
-       struct winlink  *wl;
-       int              flags, alerts;
+       struct window           *w;
+       struct session          *s, *s2;
+       struct session_group    *sg;
+       struct winlink          *wl;
+       int                      flags, alerts;

        RB_FOREACH(w, windows, &windows) {
                RB_FOREACH(s, sessions, &sessions) {
+                       if ((sg = session_group_find(s)) != NULL) {
+                               TAILQ_FOREACH(s2, &sg->sessions, gentry) {
+                                       if (s2 == s)
+                                               continue;
+                                       session_group_synchronize_from(s2);
+                               }
+                       }
+
                        RB_FOREACH(wl, winlinks, &s->windows) {
                                if (wl->window != w)
                                        continue;
```

**Comment by nicm** [comment]
I think this should be fixed now.

**Comment by MacGyverNL** [comment]
Sorry for the long silence, confirmed fixed in (at least) 2.4.
